doc/build.md

```diff
@@ -0,0 +1,43 @@
+# Build process and components
+
+APIcast is [OpenResty](https://openresty.org/en/) application. It consits of two parts: nginx configuration and lua files.
+
+## Release
+
+APIcast is released as [Docker image](https://docs.docker.com/engine/tutorials/dockerimages/). 
+
+## Dependencies
+
+APIcast uses LuaRocks to install dependencies. LuaRocks have to be properly configured,
+so it installs dependencies into correct path where OpenResty can see them. In the Docker image, rocks are installed into the application folder.
+Then this folder is added to the load path by `luarocks path`. 
+
+Lua Dependencies are defined in [`apicast-VERSION.rockspec`](https://github.com/3scale/docker-gateway/blob/50daf279b3cf2da80b20ad473ec820d7a364b688/apicast-0.1-0.rockspec) file.
+
+* `lua-resty-http` cosocket based http client to be used instead of the `ngx.location.capture`
+* `inspect` library to pretty print data structures
+* `router` used to implement internal APIs
+
+## Components
+
+APIcast is using [source-to-image](https://github.com/openshift/source-to-image) to build the final Docker image.
```

Comment:
Perhaps we should point that source-to-image should be installed before, as is a hard req or even provide a Make task for installing it..